File: sdk/py/examples/vision/ex_image.py
# OA_DOC_BEGIN: vision-image
import os

import oa

engine = oa.Engine()

asset_root = os.environ.get("OA_ASSET", "asset")
path = os.path.join(asset_root, "image/visionTestPattern320x180.jpg")

image = oa.image.decode_file(engine, path, "jpeg")
small = oa.image.resize(image, 160, 90)
adjusted = oa.image.brightness_contrast(small, 0.05, 1.1)

matrix = adjusted.as_matrix()
values = matrix.read_f32()

assert matrix.shape == [1, 3, 90, 160]
assert len(values) == 3 * 90 * 160

print(matrix.shape, min(values), max(values))
# OA_DOC_END: vision-image

File: sdk/rs/examples/vision/ex_image.rs
// OA_DOC_BEGIN: vision-image
use anyhow::Context;
use oa::{Engine, ImageFormat, Path, image};

fn main() -> anyhow::Result<()> {
	let engine = Engine::new().context("create OA engine")?;

	let path = Path::asset_rel("image/visionTestPattern320x180.jpg");

	let img = image::decode_file(&engine, &path, ImageFormat::Rgb)
		.with_context(|| format!("decode {}", path.display()))?;
	let small = image::resize(&img, 160, 90, image::InterpolationMode::Bilinear)?;
	let adjusted = image::brightness_contrast(&small, 0.05, 1.1)?;

	let matrix = adjusted.as_matrix();
	let values = matrix.read_f32()?;

	assert_eq!(matrix.shape(), [1, 3, 90, 160]);
	assert_eq!(values.len(), 3 * 90 * 160);

	let min = values.iter().copied().fold(f32::INFINITY, f32::min);
	let max = values.iter().copied().fold(f32::NEG_INFINITY, f32::max);
	println!("{:?}  min={min:.4}  max={max:.4}", matrix.shape());
	Ok(())
}
// OA_DOC_END: vision-image
